pkg/apis/storage/validation/validation_test.go

```diff
@@ -54,50 +54,53 @@ var (
+	return &storage.StorageClass{
+		ObjectMeta:    metav1.ObjectMeta{Name: "foo"},
+		Provisioner:   "kubernetes.io/foo-provisioner",
+		ReclaimPolicy: &deleteReclaimPolicy,
+	}
+}
+
+func makeStorageClassCustom(tweaks ...func(storageClass *storage.StorageClass)) *storage.StorageClass {
+	storageClass := makeValidStorageClass()
+	for _, fn := range tweaks {
+		fn(storageClass)
+	}
+	return storageClass
+}
+
+func TestValidateStorageClass(t *testing.T) {
 	retainReclaimPolicy := api.PersistentVolumeReclaimPolicy("Retain")
 	recycleReclaimPolicy := api.PersistentVolumeReclaimPolicy("Recycle")
-	successCases := []storage.StorageClass{
-		{
-			// empty parameters
-			ObjectMeta:        metav1.ObjectMeta{Name: "foo"},
-			Provisioner:       "kubernetes.io/foo-provisioner",
-			Parameters:        map[string]string{},
-			ReclaimPolicy:     &deleteReclaimPolicy,
-			VolumeBindingMode: &immediateMode1,
-		},
-		{
-			// nil parameters
-			ObjectMeta:        metav1.ObjectMeta{Name: "foo"},
-			Provisioner:       "kubernetes.io/foo-provisioner",
-			ReclaimPolicy:     &deleteReclaimPolicy,
-			VolumeBindingMode: &immediateMode1,
-		},
-		{
-			// some parameters
-			ObjectMeta:  metav1.ObjectMeta{Name: "foo"},
-			Provisioner: "kubernetes.io/foo-provisioner",
-			Parameters: map[string]string{
-				"kubernetes.io/foo-parameter": "free/form/string",
-				"foo-parameter":               "free-form-string",
-				"foo-parameter2":              "{\"embedded\": \"json\", \"with\": {\"structures\":\"inside\"}}",
-			},
-			ReclaimPolicy:     &deleteReclaimPolicy,
-			VolumeBindingMode: &immediateMode1,
-		},
-		{
-			// retain reclaimPolicy
-			ObjectMeta:        metav1.ObjectMeta{Name: "foo"},
-			Provisioner:       "kubernetes.io/foo-provisioner",
-			ReclaimPolicy:     &retainReclaimPolicy,
-			VolumeBindingMode: &immediateMode1,
-		},
+	setEmptyParametersElement := func(storageClass *storage.StorageClass) {
+		storageClass.Parameters = map[string]string{}
+	}
+	setImmediateVolumeBindingModeElement := func(storageClass *storage.StorageClass) {
+		storageClass.VolumeBindingMode = &immediateMode1
+	}
+	setMultiParametersElement := func(storageClass *storage.StorageClass) {
+		storageClass.Parameters = map[string]string{
+			"kubernetes.io/foo-parameter": "free/form/string",
+			"foo-parameter":               "free-form-string",
+			"foo-parameter2":              "{\"embedded\": \"json\", \"with\": {\"structures\":\"inside\"}}",
+		}
+	}
+	setRetrainReClaimPolicyElement := func(storageClass *storage.StorageClass) {
+		storageClass.ReclaimPolicy = &retainReclaimPolicy
+	}
+	successCases := []*storage.StorageClass{
+		makeStorageClassCustom(setEmptyParametersElement, setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setMultiParametersElement, setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setRetrainReClaimPolicyElement, setImmediateVolumeBindingModeElement),
```

Comment:
I find the `*Element` suffix redundant.

Reply:
ok, I will remove these *Element suffixes~